src/main/java/seedu/address/logic/relationship/AddRelationshipCommandParser.java

```diff
@@ -19,33 +20,57 @@ public class AddRelationshipCommandParser implements Parser<AddRelationshipComma
+                && relationKeysAndValues(relationshipMap, 1, true) != null)
+                || (relationKeysAndValues(relationshipMap, 0, true) != null
+                && relationKeysAndValues(relationshipMap, 1, true) == null)) {
+            throw new ParseException(Messages.MESSAGE_INVALID_RELATIONSHIP_COMMAND_FORMAT);
         }
-        String originUuid;
-        String targetUuid;
-        String relationshipDescriptor;
-        if (parts.length == 5) {
-            String role1 = parts[0];
-            String role2 = parts[2];
-            originUuid = ParserUtil.parseUuid(parts[1]);
-            targetUuid = ParserUtil.parseUuid(parts[3]);
-            relationshipDescriptor = parts[4];
-            if (relationshipDescriptor.equalsIgnoreCase("family")) {
+
+        String originUuid = relationKeysAndValues(relationshipMap, 0, false);
+        String targetUuid = relationKeysAndValues(relationshipMap, 1, false);
+        String relationshipDescriptor = relationKeysAndValues(relationshipMap, 2, false).toLowerCase();
+
+        if (relationKeysAndValues(relationshipMap, 0, true) != null
+                && relationKeysAndValues(relationshipMap, 1, true) != null) {
+            String role1 = relationKeysAndValues(relationshipMap, 0, true).toLowerCase();
+            String role2 = relationKeysAndValues(relationshipMap, 1, true).toLowerCase();
+
+            originUuid = ParserUtil.parseUuid(originUuid);
+            targetUuid = ParserUtil.parseUuid(targetUuid);
+            relationshipDescriptor = relationshipDescriptor.toLowerCase();
+            if (relationshipDescriptor.equals("family")) {
                 throw new ParseException("Please specify the type of familial relationship instead of 'Family'.\n"
                         + " Valid familial relations are: [bioParents, siblings, spouses]");
             }
             return new AddRelationshipCommand(originUuid, targetUuid, relationshipDescriptor, role1, role2);
         } else {
-            originUuid = ParserUtil.parseUuid(parts[0]);
-            targetUuid = ParserUtil.parseUuid(parts[1]);
-            relationshipDescriptor = parts[2];
-            if (relationshipDescriptor.equalsIgnoreCase("family")) {
+            originUuid = ParserUtil.parseUuid(originUuid);
+            targetUuid = ParserUtil.parseUuid(targetUuid);
+            relationshipDescriptor = relationshipDescriptor.toLowerCase();
+            if (relationshipDescriptor.equals("family")) {
                 throw new ParseException("Please specify the type of familial relationship instead of 'Family'.\n"
                         + " Valid familial relations are: [bioParents, siblings, spouses]");
             }
             return new AddRelationshipCommand(originUuid, targetUuid, relationshipDescriptor);
         }
     }
+
+    private String relationKeysAndValues(LinkedHashMap<String, String> relationshipMap, int index, boolean value) {
```

Comment:
Since this method is exactly the same in all three relationship Commands (add, delete and edit), maybe you could instead just put this method in relationshipUtil to not duplicate lines